lib/formatter/filter.dart

```diff
@@ -1,8 +1,8 @@
 part of angular.formatter_internal;
 
 // Too bad you can't stick typedef's inside a class.
-typedef bool _Predicate(e);
-typedef bool _Equals(a, b);
+typedef _Predicate(e);
```

Comment:
is checks for typedefs with non-dynamic argument or return types are prohibited by the dart dev compiler because the semantics are surprising.
For example:
typedef bool _Predicate(e);
Object myWackyPredicate(e) { return 42; };
print(myWackyPredicate is _Predicate);

will print true even though myWackyPredicate does not return a bool. With the new code it is at least clear exactly where the types are not specified. I do agree that it is a shame to not document that the "intent" was for _Predicate and _Equals to return bool.